test/e2e/replica_set_tls/replica_set_tls_test.go

```diff
@@ -0,0 +1,47 @@
+package replica_set_tls
+
+import (
+	"testing"
+
+	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/pkg/apis/mongodb/v1"
+	e2eutil "github.com/mongodb/mongodb-kubernetes-operator/test/e2e"
+	"github.com/mongodb/mongodb-kubernetes-operator/test/e2e/mongodbtests"
+	setup "github.com/mongodb/mongodb-kubernetes-operator/test/e2e/setup"
+	f "github.com/operator-framework/operator-sdk/pkg/test"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func TestMain(m *testing.M) {
+	f.MainEntry(m)
+}
+
+func TestReplicaSetTLS(t *testing.T) {
+	ctx, shouldCleanup := setup.InitTest(t)
+	if shouldCleanup {
+		defer ctx.Cleanup()
+	}
+
+	mdb := e2eutil.NewTestMongoDB("mdb0")
+	mdb.Spec.Security.TLS = e2eutil.NewTestTLSConfig(false)
+
+	t.Run("Create TLS Resources", mongodbtests.CreateTLSResources(&mdb, ctx))
+	t.Run("Create MongoDB Resource", mongodbtests.CreateMongoDBResource(&mdb, ctx))
+	t.Run("Config Map Was Correctly Created", mongodbtests.AutomationConfigConfigMapExists(&mdb))
+	t.Run("Stateful Set Reaches Ready State", mongodbtests.StatefulSetIsReady(&mdb))
+	t.Run("Stateful Set has OwnerReference", mongodbtests.StatefulSetHasOwnerReference(&mdb,
+		*metav1.NewControllerRef(&mdb, schema.GroupVersionKind{
+			Group:   mdbv1.SchemeGroupVersion.Group,
+			Version: mdbv1.SchemeGroupVersion.Version,
+			Kind:    mdb.Kind,
+		})))
```

Comment:
maybe we should extract this into a `defaultOwnerReference` function or similar. I think we have this call to `metav1.NewControllerRef` in several places now.

Reply:
Fixed this by refactoring all tests to use the `BasicFunctionality` test. Now this owner reference only appears there.